pkg/coscheduling/coscheduling.go

```diff
@@ -304,3 +354,40 @@ func (cs *Coscheduling) calculateWaitingPods(podGroupName, namespace string) int
 
 	return waiting
 }
+
+// markPodGroupAsExpired set the deletionTimestamp of PodGroup.
+func (cs *Coscheduling) markPodGroupAsExpired(obj interface{}) {
+	pod := obj.(*v1.Pod)
+	podGroupName, podMinAvailable, _ := GetPodGroupLabels(pod)
```

Comment:
Again, does it make sense to replace `GetPodGroupLabels` with ` getOrUpdatedPodGroupInfo` for consistency and extensibility?

Reply:
I thought about it before. We can't replace `GetPodGroupLabels ` with `getOrUpdatedPodGroupInfo `  here. Let me give you an example: 
pod1, pod2 belong to pg1.
1.pod1 is deleted. it will mark pg1 deleted.
2.pod2 is deleted. if we invoke `getOrUpdatedPodGroupInfo`, it will remove the deleted status of pod2 firstly，then In the following process, we will mark pg1 deleted again. (this operation will write sync.map again)

Note: we remove the deleted status of the same pg in `getOrUpdatedPodGroupInfo`.

Reply:
OK, it can be optimized in the future versions.